src/main/java/wolox/training/models/Book.java

```diff
@@ -78,6 +88,9 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+        Preconditions.checkNotNull(year, String.format(ErrorConstants.NOT_NULL, "year"));
+        Preconditions.checkArgument(year.matches("^(0|[1-9][0-9]*)$\n"),
```

Comment:
I recommend using this library to do the check by numbers:

import org.apache.commons.lang.StringUtils; ---> import
StringUtils.isNumeric(year);